### PR TITLE
[Refactor] Optimize creating shuffle handlers

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
[Refactor] Optimize creating shuffle handlers

### Why are the changes needed?
When creating shuffle handler, the code is too duplicate and complex.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs

### CODE REVIEW

**Comment by zuston** [comment]
PTAL @jerqi @frankliee

**Comment by jerqi** [comment]
We have the method `isMemoryShuffleEnabled` in the MR client and Spark client, you can modify them, too.

File: storage/src/main/java/org/apache/uniffle/storage/util/StorageType.java [hunk 1]
```diff
@@ -18,10 +18,33 @@
...
-  MEMORY_LOCALFILE,
-  MEMORY_HDFS,
-  MEMORY_LOCALFILE_HDFS
+  MEMORY(1),
+  LOCALFILE(2),
+  HDFS(4),
+  LOCALFILE_HDFS(6),
+  MEMORY_LOCALFILE(3),
+  MEMORY_HDFS(5),
+  MEMORY_LOCALFILE_HDFS(7);
```
Comment (frankliee) [inline storage/src/main/java/org/apache/uniffle/storage/util/StorageType.java:27]: Why this order？

Comment (zuston) [reply storage/src/main/java/org/apache/uniffle/storage/util/StorageType.java:27]: Just follow the original order

**Comment by zuston** [comment]
> We have the method `isMemoryShuffleEnabled` in the MR client and Spark client, you can modify them, too.

Done

**Comment by jerqi** [comment]
> > We have the method `isMemoryShuffleEnabled` in the MR client and Spark client, you can modify them, too.
> 
> Done

You only modify the spark2 client. How about spark3 and mr?

**Comment by zuston** [comment]
> > > We have the method `isMemoryShuffleEnabled` in the MR client and Spark client, you can modify them, too.
> > 
> > 
> > Done
> 
> You only modify the spark2 client. How about spark3 and mr?

Updated. I forgot it.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston @frankliee , wait for CI

**Comment by zuston** [comment]
Flaky test: https://github.com/apache/incubator-uniffle/actions/runs/3225484333/jobs/5277849404

**Comment by zuston** [comment]
Do u mind merging this ? @jerqi

**Comment by jerqi** [comment]
> Do u mind merging this ? @jerqi

merged.

**Pull request merged**